x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_results.tsx

```diff
@@ -33,10 +50,66 @@ export const useLogAnalysisResults = ({
     getLogEntryRate();
   }, [sourceId, startTime, endTime, bucketDuration]);
 
+  const results: Results | null = useMemo(() => {
+    if (logEntryRate) {
+      return {
+        bucketDuration: logEntryRate.bucketDuration,
+        totalNumberOfLogEntries: logEntryRate.totalNumberOfLogEntries,
+        histogramBuckets: logEntryRate.histogramBuckets,
+        partitionBuckets: formatLogEntryRateResultsByPartition(logEntryRate),
+      };
+    } else {
+      return null;
+    }
+  }, [logEntryRate]);
+
   return {
     isLoading,
-    logEntryRate,
+    results,
   };
 };
 
 export const LogAnalysisResults = createContainer(useLogAnalysisResults);
+
+const formatLogEntryRateResultsByPartition = (
+  results: GetLogEntryRateSuccessResponsePayload['data']
+): PartitionRecord => {
+  const partitionedBuckets = results.histogramBuckets.reduce<
+    Record<string, { buckets: PartitionBucket[] }>
+  >((partitionResults, bucket) => {
+    return bucket.partitions.reduce<Record<string, { buckets: PartitionBucket[] }>>(
+      (_partitionResults, partition) => {
+        return {
```

Comment:
How many partitions are there usually?

This returns a new object for every partition. For small numbers should be fine (I prefer the spread syntax), but if the number is large-ish I'd change this to 

```
partitionResults[partition.partitionId] = { ... }
return partitionResults;
```

to avoid the overhead

Reply:
Usually a small number. Tanya said the following based on telemetry information when partitioning was first being discussed: "Telemetry data shows that it's usually just a few and almost always under 10.".